### DISCUSSION TITLE
[@types/update-notifier] Breaks with ESM

### DISCUSSION THREAD

**Discussion by Zamiell** [DefinitelyTyped/DefinitelyTyped#64159] [Issues with a @types package] [answered]
When using update-notifier in a brand new TypeScript project with ESM, it uses `any` because the import resolution breaks, which will cause the "@typescript-eslint/no-unsafe-member-access" / "@typescript-eslint/no-unsafe-assignment" ESLint rules to be triggered.

I think the problem is because of this line in the "index.d.ts" file provided by the "@types/update-notifier" package:

```ts
import UpdateNotifier, { NotifyOptions, Package, Settings, UpdateInfo } from './update-notifier';
```

I'm not sure of the correct fix, but I do know that it should be definitely possible to construct a "index.d.ts" file that willl work in both commonjs and ESM projects.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
the DT module is fully ESM, it has no CJS version at v6.
What do you mean by brand new ESM project setup, thanks!

this (JS) one was built with ESNext setup in tsconfig:
```js
import updateNotifier from 'update-notifier';
import packageJson from './package.json' assert { type: 'json' };
updateNotifier({ pkg: packageJson }).notify();
```

**Comment by Zamiell** [comment]
@peterblazejewicz See my reproduction here: https://github.com/Zamiell/test-3#test-3

**Reply by peterblazejewicz** [reply]
you're wright, the proper ESM file import was missing, so the file was not published (and it's missing on local install),
fix: #64183 64183

**Reply by Zamiell** [reply]
Thanks peter.

**Comment by peterblazejewicz** [comment, marked as answer]
fixed in #64183
